### ISSUE TITLE
Mobs killed by turrets don't drop loot and xp.

### ISSUE THREAD

**Issue by Pabilo8** [BluSunrize/ImmersiveEngineering#3128]
Description of the issue:
[I really don't know if this is intended or not, so I won't label it as a suggestion ^^]
When using a turret (both gun and chemthrower) mobs are killed but no item or xp drops. However if I hit a mob, and then it's then killed by turret it has a normal drop chance (both items and xp).

Turrets would be very nice way of automating grinders (I tried to transfer water with endertanks mod to the nether to make a blaze farm with chemthrowers).

Versions & Modlist
Using MC 1.12.2
Forge 14.23.4.2738
IE 0.12-84
Full Modlist: https://pastebin.com/g8e4YKv8

**Comment by malte0811** [comment]
This makes sense for XP, it only drops when something is killed by a player (MC defines "killed by a player" as "has been hit by a player in the last 5 seconds" or something like that IIRC). What item drops did you test with? Blazes or something else as well?

**Comment by Pabilo8** [comment]
Turns out that only blazes don't drop their items. I tested zombies, skeletons, cave spiders and blazes in both overworld and the nether with both turrets.

**Comment by BluSunrize** [comment]
Yeah blazes have their drops relying on a player kill. That's intentional vanilla behavior. Our turrets don't really emulate player behavior, nor have I yet considered that they should...

**Comment by PrincessRTFM** [comment]
I kind of feel like they _shouldn't_, personally. I don't feel like these turrets are meant for mob grinding, they're for base defense and protecting restricted areas. In my opinion, you don't set up turrets to farm mobs, you set them up to stop invasions. Just my two cents.

**Comment by Pabilo8** [comment]
I fixed my problem with [this mod](https://minecraft.curseforge.com/projects/mob-drops).

Re: @PrincessRTFM 
I agree, they shouldn't, but currently I really don't find a use for the turrets (and the razor wire) other than grinding. Mobs don't swarm the player's base and on my server everyone is working together, so no pvp usage either. The only usage I found for them (in terms of base defence) is to protect my factory in night with a [Blood Moon](https://minecraft.curseforge.com/projects/bloodmoon).
